docs/deployment-patterns/deployment-as-code.md

```diff
@@ -0,0 +1,113 @@
+1. **Push configuration changes to Octopus project treating the code as a source of truth**
+1. Push application packages to deployment feed
+1. Create a release in Octopus
+1. Deploy the release via Octopus to your Dev/Test environment
+1. Promote the release to other environments
+
+## Consistency and repeatability
+
+When you move to manage your Octopus configuration as code, Octopus still makes sure your deployments are consistent and repeatable. Whenever you push a configuration change to your Octopus project via code, it's just like people using the user interface or API to make changes. When you create a release, Octopus takes a snapshot of the deployment process, variables, and packages making every deployment of that release consistent and repeatable, regardless of whether the project was configured by a person or by code.
+
+This means you can push configuration changes to Octopus as code, and get the same consistent and repeatable experience you expect for your deployments.
+
+### Introducing changes safely
+
+This is where things get a bit more difficult compared to the [scripts as code](#scripts-as-code) pattern.
+
+As a starting point, we recommend **pushing configuration changes from one specific branch into Octopus.** Using the example of git, you should only push changes from the `master` branch into Octopus, and use [Channels](/docs/deployment-process/channels/index.md) to safely introduce changes to your process and variables.
+
+The general process should look something like this, tailored to your situation:
+
+1. Create a channel to match your branch, with package version rules to enforce the integrity of the release process. _You can create channels manually, or automatically as part of your build pipeline if that suits._
+1. Make the changes on your branch, making sure to scope your changes to your channel to avoid interrupting deployments from other branches. _You can scope each step/action and variable value to a specific channel for isolation._
+1. Get a peer to review your configuration change on your branch.
+1. Merge your configuration change to the `master` branch so your changes are actually pushed into Octopus.
+1. Test your change by deploying releases through your channel.
+  a. If you are unhappy with your change, fix it in your branch, get a peer to review your new commits, merge the new commits to `master`, and repeat your testing.
+1. Once you are happy the changes are safe to share:
+  a. Remove the channel scoping from your changes in your branch.
+  b. Get a peer to review this final change.
+  c. Merge your changes to the `master` branch.
+  d. Test your brand new deployment through the `master` branch and your main channel.
+  e. If something goes wrong, you can revert the single commit, isolating your changes back to your channel.
+
+:::hint
+Are you finding this approach annoying/troublesome/tedious? We would like to [talk with you about your dream scenario](https://octopus.com/support)!
```

Comment:
Alternate wording: If you have thoughts about how deployment as code could better support your organization, we would like...